### PR TITLE
Elect-359 Moving logic for bucket reconnects into library.

### PR DESCRIPTION
**Pull request by binaryalchemist**

### CODE REVIEW

File: test/mocks/mock-turncoat.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
```
Comment (enigmatic00) [inline test/mocks/mock-turncoat.js]: This mock isn't used

File: test/unit/lachaise-tests.js [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+        var lac = new LaChaise(config);
+
+        lac.bucket('test', function(err, bucket) {});
+
+        bucketMock.connected = true;
+        bucketMock.emit('connect');
+
+        lac.bucket('test', function(err, bucket) {
+            expect(bucket).to.be.an.instanceOf(EventEmitter);
```
Comment (enigmatic00) [inline test/unit/lachaise-tests.js]: given the description, the assertion should be `expect(bucket.connected).to.be.true`

File: test/unit/lachaise-tests.js [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+            done();
+        });
+    });
+
+    it('should retry when it can\'t open a bucket', function(done) {
+        var spyOpenBucket = sinon.stub();
+        var mockOpenBucket = new EventEmitter;
+        mockOpenBucket.connected = true;
+
+        var mockCouchbase = function() {
```
Comment (enigmatic00) [inline test/unit/lachaise-tests.js]: This doesn't need to be a function. Could be re-written as 
```
var mockCouchbase = return {
     Cluster: function(){
         return {
                 openBucket: spyOpenBucket;
         }
     }
}
```
then used as
`mockery.registerMock('couchbase', mockCouchbase);`

File: test/unit/lachaise-tests.js [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+
+        lac.bucket('test', function() {
+            expect(spyOpenBucket.callCount).to.equal(2);
+            done();
+        });
+
+        mockOpenBucket.emit('error', { msg: 'retry-able', code: 16  });
+    });
+
+    it('should return on unrecoverable errors.', function(done) {
```
Comment (enigmatic00) [inline test/unit/lachaise-tests.js]: `it('should return an error on unrecoverable errors'...`

File: test/unit/lachaise-tests.js [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+            expect(err.msg).to.equal('evil');
+            done();
+        });
+
+        lac.bucket('test', function() {});
+
+        mockOpenBucket.emit('error', { msg: 'evil', code: 666 });
+    });
+
+    it('should retry when it can\'t open a bucket', function(done) {
```
Comment (enigmatic00) [inline test/unit/lachaise-tests.js]: this appears to be a duplicate of the test on line 92 but with more assertions

File: README.md [hunk 6]
```diff
@@ -1,2 +1,22 @@
...
+# Usage Example
+
+```Node
+var LaChaise = require('lachaise');
+
+//config declaration omitted for brevity
+
+var lac = new LaChaise(config);
+
+lac.bucket('my_couchbase_bucket', function(bucket) {
```
Comment (enigmatic00) [inline README.md]: Shouldn't this be `+lac.bucket('my_couchbase_bucket', function(err, bucket) {` ?

File: test/unit/lachaise-tests.js (hunk 2 above)
Comment (binaryalchemist) [inline test/unit/lachaise-tests.js]: I actually should have both...

**Pull request merged**